### PR TITLE
Sol 3 Keyboard from RGBKB

### PR DESCRIPTION
**Pull request by XScorpion2**
Sol 3 Keyboard from RGBKB

## Description
This adds QMK Keyboard support for the RGBKB Sol 3. 
The Default keymap version is coming factory flashed on the Sol 3.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rgbkb/sol3/keymaps/kageurufu/config.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// No need for the single versions when multi performance isn't a problem =D
+#define DISABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
+#define DISABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
+#define DISABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
+#define DISABLE_RGB_MATRIX_SPLASH
+#define DISABLE_RGB_MATRIX_SOLID_SPLASH
+
+// 20m timeout (20m * 60s * 1000mil)
+// #define RGB_DISABLE_TIMEOUT 1200000
+#define RGB_DISABLE_WHEN_USB_SUSPENDED true
```
Comment (drashna) [inline keyboards/rgbkb/sol3/keymaps/kageurufu/config.h:13]:
```suggestion
#define RGB_DISABLE_WHEN_USB_SUSPENDED
```

File: keyboards/rgbkb/sol3/keymaps/kageurufu/keymap.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/rgbkb/sol3/keymaps/kageurufu/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/rgbkb/sol3/rules.mk [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+AUDIO_ENABLE = yes          # Audio output
+AUDIO_DRIVER = dac_basic
```
Comment (drashna) [inline keyboards/rgbkb/sol3/rules.mk:19]: I'd recommend dac_additive instead.  dac basic seems to have some issues, especially for stuff like music mode and audio clicky. 
```suggestion
AUDIO_DRIVER = dac_additive
```

File: keyboards/rgbkb/sol3/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+}
+
+static void render_touch(void)
+{
+    // Host Touch LED Status
+    oled_write_P(!touch_encoder_toggled() ? PSTR("TOUCH")  : PSTR("     "), false);
+    oled_write_P(touch_encoder_calibrating() ? PSTR("CLBRT")  : PSTR("     "), false);
+}
+
+bool oled_task_user(void) {
```
Comment (drashna) [inline keyboards/rgbkb/sol3/keymaps/default/keymap.c:226]: Please move/add default oled behavior using `oled_render_kb` to the keyboard code.

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by Legonut** [comment]
Tested and working on my Sol 3, approved by RGBKB (me)

**Review by drashna — APPROVED** [review]

File: keyboards/rgbkb/sol3/rules.mk [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+AUDIO_ENABLE = yes          # Audio output
+AUDIO_DRIVER = dac_additive
+
+DYNAMIC_MACRO_ENABLE = yes
+DIP_SWITCH_ENABLE = yes
+
+WS2812_DRIVER = pwm
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
```
Comment (fauxpark) [inline keyboards/rgbkb/sol3/rules.mk:25]:
```suggestion
RGBLIGHT_ENABLE = no
```

File: keyboards/rgbkb/sol3/keymaps/kageurufu/keymap.c [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+      ________________FUNCTION_L4________________, _______,                   _______, ________________FUNCTION_R4________________, \
+      ________________FUNCTION_L5________________, ADJ,     KC_DEL,  _______, ADJ,     ________________FUNCTION_R5________________  \
+      ),
+
+  [_ADJ] = EXPAND_LAYOUT( \
+      _________________ADJUST_L1_________________, _______,                   _______, _________________ADJUST_R1_________________, \
+      _________________ADJUST_L2_________________, _______,                   _______, _________________ADJUST_R2_________________, \
+      _________________ADJUST_L3_________________, _______,                   _______, _________________ADJUST_R3_________________, \
+      _________________ADJUST_L4_________________, _______,                   _______, _________________ADJUST_R4_________________, \
+      _________________ADJUST_L5_________________, _______, _______, _______, _______, _________________ADJUST_R5_________________  \
```
Comment (fauxpark) [inline keyboards/rgbkb/sol3/keymaps/kageurufu/keymap.c:58]:
```suggestion
  [_QWERTY] = EXPAND_LAYOUT(
      KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    RGB_MOD,                   KC_EQL,  KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
      KC_TAB,  _________________QWERTY_L1_________________, KC_LBRC,                   KC_RBRC, _________________QWERTY_R1_________________, KC_BSLS,
      FN_ESC,  _________________QWERTY_L2_________________, RGB_SAI,                   RGB_VAI, _________________QWERTY_R2_________________, KC_QUOT,
      KC_LSPO, _________________QWERTY_L3_________________, RGB_SAD,                   RGB_VAD, _________________QWERTY_R3_________________, KC_RSPC,
      KC_LCTL, KC_LGUI, KC_LGUI, KC_LALT, FN,      KC_SPC,  KC_MUTE, KC_BSPC, KC_ENT,  FN,      KC_SPC,  KC_MINS, KC_EQL,  KC_DOWN, KC_PGUP, KC_PGDN
  ),

  [_FN] = EXPAND_LAYOUT(
      ________________FUNCTION_L1________________, _______,                   KC_PSCR, ________________FUNCTION_R1________________,
      ________________FUNCTION_L2________________, _______,                   _______, ________________FUNCTION_R2________________,
      ________________FUNCTION_L3________________, _______,                   _______, ________________FUNCTION_R3________________,
      ________________FUNCTION_L4________________, _______,                   _______, ________________FUNCTION_R4________________,
      ________________FUNCTION_L5________________, ADJ,     KC_DEL,  _______, ADJ,     ________________FUNCTION_R5________________
      ),

  [_ADJ] = EXPAND_LAYOUT(
      _________________ADJUST_L1_________________, _______,                   _______, _________________ADJUST_R1_________________,
      _________________ADJUST_L2_________________, _______,                   _______, _________________ADJUST_R2_________________,
      _________________ADJUST_L3_________________, _______,                   _______, _________________ADJUST_R3_________________,
      _________________ADJUST_L4_________________, _______,                   _______, _________________ADJUST_R4_________________,
      _________________ADJUST_L5_________________, _______, _______, _______, _______, _________________ADJUST_R5_________________
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by XScorpion2** [comment]
Code review changes applied @fauxpark

File: keyboards/rgbkb/sol3/rules.mk [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+# Touch encoder needs
+SRC += ../common/touch_encoder.c
+SRC += ../common/common_oled.c
+QUANTUM_LIB_SRC += i2c_master.c
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Enable Bootmagic Lite
```
Comment (fauxpark) [inline keyboards/rgbkb/sol3/rules.mk:15]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by XScorpion2** [comment]
Bump, can this land before the breaking change merge?

**Review by fauxpark — APPROVED** [review]
`.noci` file can be deleted since Travis is no longer running.

**Comment by drashna** [comment]
Thanks!

**Comment by XScorpion2** [comment]
no thank you @drashna and @fauxpark!

**Pull request merged**